cloud_storage_handler/main.py

```diff
@@ -39,13 +32,56 @@ def init_app() -> FlaskApp:
 
     foca = Foca(
         config_file=config_path,
+        custom_config_model="cloud_storage_handler.custom_config.CustomConfig",
     )
-    return foca.create_app()
+    foca_config = foca.conf
+    try:
+        custom_config_data = foca_config.custom.dict()
+        custom_config = CustomConfig(**custom_config_data)
+    except ValidationError as e:
+        logger.error(f"Error parsing custom configuration: {e}")
+        raise
+    minio_config = custom_config.minio
+
+    # Create the Connexion FlaskApp instance
+    return foca.create_app(), minio_config
 
 
 def main() -> None:
     """Run FOCA application."""
-    app = init_app()
+    try:
+        app, minio_config = init_app()
+    except Exception as e:
+        logger.error(f"Unexpected error during initialization: {e}")
+        sys.exit(1)
+    foca_app = app.app
+
+    # Initialize MinIO client
+    try:
+        minio_client = Minio(
+            endpoint=f"{minio_config.hostname}:{minio_config.port}",
+            access_key=minio_config.access_key,
+            secret_key=minio_config.secret_key,
+            secure=minio_config.secure,
+        )
+        foca_app.config["minio_client"] = minio_client
+    except Exception as e:
+        logger.error(f"Error initializing MinIO client: {e}")
+        return
```

Comment:
Such error handling is really not particularly helpful, just let `Minio()` raise the error...